src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -486,39 +571,42 @@ public Solution AddProjectReferences(ProjectId projectId, IEnumerable<ProjectRef
         /// Create a new solution instance with the project specified updated to no longer
         /// include the specified project reference.
         /// </summary>
+        /// <exception cref="ArgumentNullException"><paramref name="projectId"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="projectReference"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException">The solution does not contain <paramref name="projectId"/>.</exception>
         public Solution RemoveProjectReference(ProjectId projectId, ProjectReference projectReference)
         {
-            var newState = _state.RemoveProjectReference(projectId, projectReference);
-            if (newState == _state)
+            if (projectReference == null)
             {
-                return this;
+                throw new ArgumentNullException(nameof(projectReference));
             }
 
-            return new Solution(newState);
-        }
+            CheckContainsProject(projectId);
 
-        /// <summary>
-        /// Create a new solution instance with the project specified updated to contain
-        /// the specified list of project references.
-        /// </summary>
-        public Solution WithProjectReferences(ProjectId projectId, IEnumerable<ProjectReference> projectReferences)
-        {
-            var newState = _state.WithProjectReferences(projectId, projectReferences);
+            var newState = _state.RemoveProjectReference(projectId, projectReference);
             if (newState == _state)
             {
-                return this;
+                throw new ArgumentException(WorkspacesResources.Project_does_not_contain_specified_reference, nameof(projectReference));
             }
 
             return new Solution(newState);
         }
 
         /// <summary>
-        /// Creates a new solution instance with the project documents in the order by the specified document ids.
-        /// The specified document ids must be the same as what is already in the project; no adding or removing is allowed.
+        /// Create a new solution instance with the project specified updated to contain
+        /// the specified list of project references.
         /// </summary>
-        public Solution WithProjectDocumentsOrder(ProjectId projectId, ImmutableList<DocumentId> documentIds)
+        /// <param name="projectId">Id of the project whose references to replace with <paramref name="projectReferences"/>.</param>
+        /// <param name="projectReferences">New project references.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="projectId"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="projectReferences"/> contains <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="projectReferences"/> contains duplicate items.</exception>
```

Comment:
Ooh we're even documenting behavior. What madness is this!

Reply:
Just a little bit :)